SwanGallery/src/index.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+
+import { downloadUrlFromServer } from './handler';
+
+import { Dialog, ICommandPalette } from '@jupyterlab/apputils';
+
+import { IFrame, MainAreaWidget } from '@jupyterlab/apputils';
+import { ILauncher } from '@jupyterlab/launcher';
+import { swanGalleryIcon } from './icons';
+
+import { showDialog } from "@jupyterlab/apputils";
+import { ConfigSection } from '@jupyterlab/services';
+
+/*
+  Fetch URL of the gallery website from the jupyter server config
+*/
+async function initConfigurationFromServer() {
+  const galleryUrl = await ConfigSection.create({
+    name: 'gallery',
+  });
+  return galleryUrl.data?.gallery_url as string;
+}
+
+async function activate(
+  app: JupyterFrontEnd,
+  palette: ICommandPalette,
+  launcher: ILauncher
+): Promise<void> {
+  console.log('JupyterLab extension SwanGallery is activated!');
+  
+  const galleryUrl = await initConfigurationFromServer();
+  
+  let flag = true;
+  const command = 'swangallery:open';
+  app.commands.addCommand(command, {
+    label: 'SWAN Gallery',
+    icon: swanGalleryIcon,
+    execute: () => {
+      const content = new IFrame({
+        sandbox: [
+          'allow-scripts',
+          'allow-same-origin',
+          'allow-modals',
+          'allow-downloads', // Required to download notebooks
+          'allow-popups' // Required for opening external links in new tabs
+        ]
+      });
+
+      // Avoid multiple executes of the same event
+      if (flag === true) {
+        flag = false;
+        window.addEventListener('message', async (event: any) => {
+          if (event.origin != galleryUrl) {
```

Comment:
Yeah, if the origin matches, then I think we can trust the URL. 
If we need this check, it should perhaps be implemented in the ContentsManager/server side which does the download. It would be a filter for trusted repositories/URLs (github, cernbox, eos, etc URLs).

Some of the links in the gallery itself are external URLs (for downloading entire repos)